pkg/network/render.go

```diff
@@ -800,16 +801,32 @@ func renderMultiNetworkpolicy(conf *operv1.NetworkSpec, manifestDir string) ([]*
 }
 
 // renderNetworkDiagnostics renders the connectivity checks
-func renderNetworkDiagnostics(conf *operv1.NetworkSpec, manifestDir string) ([]*uns.Unstructured, error) {
-	if conf.DisableNetworkDiagnostics {
+func renderNetworkDiagnostics(operConf *operv1.NetworkSpec, clusterConf *configv1.NetworkSpec, manifestDir string) ([]*uns.Unstructured, error) {
+	// network diagnostics feature is disabled when clusterConf.NetworkDiagnostics.Mode is set to "Disabled"
+	// or when clusterConf.NetworkDiagnostics is empty and the legacy operConf.DisableNetworkDiagnostics is true
+	if clusterConf.NetworkDiagnostics.Mode == configv1.NetworkDiagnosticsDisabled ||
+		reflect.DeepEqual(clusterConf.NetworkDiagnostics, configv1.NetworkDiagnostics{}) && operConf.DisableNetworkDiagnostics {
```

Comment:
if Mode is empty, where are you defaulting it?

Reply:
If the mode is empty and `operConf.DisableNetworkDiagnostics` is not set then we render the diagnostics.
Empty is an allowed value in the API, from the API docs:
```
 this means the user has no opinion and the platform is left
 to choose reasonable defaults. These defaults are subject to change over time.
 The current default is All.
```